### ISSUE TITLE
Assignee syncing overwrites assignee on JIRA with "unassigned" when it can't match the assignee

### ISSUE THREAD

**Issue by gijsk** [mozilla/jira-bugzilla-integration#229]
STR:

1. assign bug to me (I use a gmail bugzilla address)
2. look at JIRA, realize that JBI doesn't know who I am so assign manually in JIRA to my moco address
3. Update the bug (e.g. by landing the patch)

ER:
assignee stays the same in JIRA

AR:
assignee gets cleared in JIRA

This is obviously a huge personal affront to my dignity and needs to be fixed ASAP!!!111 one eleven. 😉

(More seriously, it's not super important to me personally but I guess for other logic it might be nice if we didn't do this.)

**Comment by Mossop** [comment]
I intentionally made it clear the assignee when it couldn't find you in Jira though maybe that is a mistake. But in theory JBI should only update the assignee in Jira when the assignee in Bugzilla changes so what you see in step three isn't intentional. This all happens here: https://github.com/mozilla/jira-bugzilla-integration/blob/main/jbi/actions/default_with_assignee_and_status.py#L82

**Comment by gijsk** [comment]
> But in theory JBI should only update the assignee in Jira when the assignee in Bugzilla changes so what you see in step three isn't intentional. This all happens here: https://github.com/mozilla/jira-bugzilla-integration/blob/main/jbi/actions/default_with_assignee_and_status.py#L82

Ah, hm. So maybe the issue here is actually that I ended up updating the JIRA ticket *before* updating the bmo ticket (or at least before the bmo change made it to JIRA)? Which I guess _sort of_ makes this bug invalid?

**Comment by gijsk** [comment]
The related issue is that it would be nice if we had some kind of mapping here from bmo address to phab username to moco/jira account... but I don't know if we can easily pull that data from somewhere rather than duplicating it again.

**Comment by Mossop** [comment]
Yeah the times of the changes are so close that my best guess is you set the assignee in Jira moments before JBI came along to do its thing

**Comment by leplatrem** [comment]
### STR

Users annoyed by this bug:

1. Bugzilla ticket has no assignee 
2. Dev works on it, assign the Jira issue to themselves
3. Dev attaches patch on Bugzilla ticket
4. Bugzilla sets assignee to author automatically
5. JBI clears the Jira issue assignee to nobody


### Event received

```
'[{"field":"assigned_to","added":"[xxx+bmo@mozilla.com](mailto:xxx+bmo@mozilla.com)","removed":"[nobody@mozilla.org](mailto:nobody@mozilla.org)"},{"added":"ASSIGNED","removed":"NEW","field":"status"}]
```

```
Handling incoming request
'Getting issue FIDEFE-6450
'Received issue FIDEFE-6450
Execute action 'fidefe' for Bug 1952962
'Find Jira user with email [xxx+bmo@mozilla.com](mailto:xxx+bmo@mozilla.com)
'User [xxx+bmo@mozilla.com](mailto:xx+bmo@mozilla.com) not found
Clearing assignee
Updating Jira status to In Progress
Action 'fidefe' executed successfully for Bug 1952962
```
